resource/opentelemetry-resource-detector-azure/pyproject.toml

```diff
@@ -26,7 +26,7 @@ classifiers = [
 ]
 dependencies = [
   "opentelemetry-sdk ~= 1.21",
-  "opentelemetry-instrumentation ~= 0.43b0",
+  "opentelemetry-instrumentation ~= 0.44b0",
```

Comment:
Did we introduce anything in 0.44b0 that the resource detector relies on?

Reply:
yep, `suppress_instrumentation` in `opentelemetry.instrumentation.utils`

```
______________________________________________ ERROR collecting resource/opentelemetry-resource-detector-azure/tests/test_vm.py _______________________________________________ 
ImportError while importing test module 'D:\repo\opentelemetry-python-contrib\resource\opentelemetry-resource-detector-azure\tests\test_vm.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.12_3.12.2032.0_x64__qbz5n2kfra8p0\Lib\importlib\__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
resource\opentelemetry-resource-detector-azure\tests\test_vm.py:18: in <module>
    from opentelemetry.resource.detector.azure.vm import AzureVMResourceDetector
resource\opentelemetry-resource-detector-azure\src\opentelemetry\resource\detector\azure\__init__.py:20: in <module>
    from .vm import AzureVMResourceDetector
resource\opentelemetry-resource-detector-azure\src\opentelemetry\resource\detector\azure\vm.py:20: in <module>
    from opentelemetry.instrumentation.utils import suppress_instrumentation
E   ImportError: cannot import name 'suppress_instrumentation' from 'opentelemetry.instrumentation.utils' (D:\repo\opentelemetry-python-contrib\.tox\py312-test-resource-detector-azure\Lib\site-packages\opentelemetry\instrumentation\utils.py)
!!
```